apps/web/components/AdditionalCalendarSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import Select from "react-select";
+import { OptionProps } from "react-select";
+
+import { InstallAppButton } from "@calcom/app-store/components";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+import { Button } from "@calcom/ui";
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
+  return (
+    <InstallAppButton
+      type={data.type}
+      render={(props) => (
+        <Button {...props} color="minimal" className="w-full">
+          {/* eslint-disable @next/next/no-img-element */}
+          {data.image && <img className="float-left mr-3 inline h-5 w-5" src={data.image} alt={data.label} />}
+          <p>{data.label}</p>
+        </Button>
+      )}
+    />
+  );
+};
+
+type AppOutput = inferQueryOutput<"viewer.integrations">["items"][0];
+
+const AdditionalCalendarSelector = ({ isLoading }: Props): JSX.Element | null => {
+  const { t } = useLocale();
+  const query = trpc.useQuery(["viewer.integrations", { variant: "calendar" }]);
+  const installedFilter = (app: AppOutput) => app.credentialIds.length > 0 || app.isGlobal;
+
+  return (
+    <QueryCell
+      query={query}
+      success={({ data }) => {
+        const options = data.items.filter(installedFilter).map((item) => ({
+          value: item.slug,
+          label: item.name,
+          image: item.imageSrc,
+          type: item.type,
+        }));
+        const customStyles = {
+          // Forgive me father for I have sinned, "any" is necessary here
```

Comment:
😇

Reply:
I think it would be good we use this comment to explain why any it's necessary here so in the future if another one takes up this code can understand easily what its happening here

Reply:
I actually tried to make it not `any` but I couldn't find a type to cover it. This is an object from `react-select` library to handle custom styles over their exported Select element. It's not that important, but I'm happy to include that description if we think it helps.

Reply:
There's no benefit extracting the variable here. Addressed on my other comment.

Reply:
Gotcha!